### ISSUE TITLE
dateutil does not accept a dt argument of None

### ISSUE THREAD

**Issue by labrys** [dateutil/dateutil#132]
Per the Python Standard Library Documentation in regards to a `tzinfo` objects `tzname(...)`, `dst(...)`, and `utcoffset(...)` methods:

> These methods are called by a datetime or time object, in response to their methods of the same names.  A datetime object passes itself as the argument, and a time object passes None as the argument. A tzinfo subclass’s methods should therefore be prepared to accept a dt argument of None, or of class datetime.

**Comment by pganssle** [comment]
Confirmed. Below is a MWE:

```python
>>> from dateutil import tz
>>> from datetime import time
>>> import pytz
>>> 
>>> td = time(13, 20, tzinfo=tz.gettz('Europe/Minsk'))
>>> tp = time(13, 20, tzinfo=pytz.timezone('Europe/Minsk'))
>>> 
>>> print(tp.utcoffset())
None
>>> print(td.utcoffset())
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python2.7/dist-packages/dateutil/tz.py", line 447, in utcoffset
    return self._find_ttinfo(dt).delta
  File "/usr/lib/python2.7/dist-packages/dateutil/tz.py", line 423, in _find_ttinfo
    + dt.second)
AttributeError: 'NoneType' object has no attribute 'toordinal'
```
I'm assuming here that `pytz` has the correct behavior and this should return `None`.

**Comment by labrys** [comment]
`tz.tzlocal()` and `tz.tzwinlocal()` both fail for each method `tzname(dt)`, `dst(dt)`, `utcoffset(dt)`

**Comment by pganssle** [comment]
Fixed in #181.

**Comment by labrys** [comment]
Confirmed working on Windows with various python versions.
